### PR TITLE
Adding check UUID format for cluster name in HTTP requests

### LINKED ISSUE #69: Check if cluster name in REST API requests is correct GUID
Possibly useful libraries to use:
https://github.com/satori/go.uuid
https://godoc.org/github.com/google/uuid

### PR DESCRIPTION
**Pull request by JoseLSegura**
# Description

Checking UUID format on the cluster name for the get reports HTTP requests.

I added some unit tests for the server package. IDK if it is aligned with the rest of the unit tests or even if it is well done because this are my first unit tests in Go

Fixes #69 

## Type of change

- New feature (non-breaking change which adds functionality)
- Unit tests (no changes in the code)

## Testing steps
Usual "make test"

### CODE REVIEW

File: server/server.go [hunk 1]
```diff
@@ -131,7 +132,13 @@ func (server HTTPServer) readClusterName(writer http.ResponseWriter, request *ht
 		responses.SendInternalServerError(writer, message)
 		return types.ClusterName(""), errors.New(message)
 	}
-	// TODO: add check for GUID-like name
+
+	if _, parseErr := uuid.Parse(clusterName); parseErr != nil {
```
Comment (tisnik) [inline server/server.go]: just `err` is IMHO more idiomatic in Go

File: server/server.go [hunk 2]
```diff
@@ -151,7 +158,7 @@ func (server HTTPServer) listOfClustersForOrganization(writer http.ResponseWrite
 	}
 }
 
-func (server HTTPServer) readReportForCluster(writer http.ResponseWriter, request *http.Request) {
+func (server HTTPServer) ReadReportForCluster(writer http.ResponseWriter, request *http.Request) {
```
Comment (tisnik) [inline server/server.go]: this need not be exported (ie. publicly visible)

Comment (tisnik) [reply server/server.go]: and `goling` basically said the same on CI :)

Comment (JoseLSegura) [reply server/server.go]: Oh, sorry. I pushed it by mistake. Initially I was using a different (and not productive at all) approach for the UT and I tried to use this function directly, so I needed to change the visibility and forget to get to its correct visibility

File: server/server.go (hunk 1 above)
Comment (JoseLSegura) [reply server/server.go]: Ok

**Pull request merged**